swift/extractor/infra/SwiftLocationExtractor.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <swift/AST/SourceFile.h>
+#include <swift/Basic/SourceManager.h>
+
+#include "swift/extractor/trap/TrapDomain.h"
+#include "swift/extractor/trap/generated/TrapEntries.h"
+#include "swift/extractor/trap/generated/TrapClasses.h"
+#include "swift/extractor/infra/file/PathHash.h"
+
+namespace codeql {
+
+class TrapDomain;
+
+class SwiftLocationExtractor {
```

Comment:
nit: as this class has no template member function (:tada:), we could split all its member functions definitions into a separate cpp file, which is a bit friendlier for the build